nixos/modules/virtualisation/nixos-containers.nix

```diff
@@ -471,13 +471,14 @@ in
               '';
               type = let
                 confPkgs = if config.pkgs == null then pkgs else config.pkgs;
+                evalConfigArgs = optionalAttrs (config.pkgs != null) {
+                  pkgs = confPkgs;
+                  inherit (confPkgs) lib;
+                };
```

Comment:
We should probably write some code to warn users that when defining your own `pkgs` for a container, no overlays will apply.

Reply:
I think the extra description of `pkgs` is enough.
Maybe we should revert to the old behaviour, to avoid breaking existing configs. The only downside is the decreased performance.